xarray/plot/plot.py

```diff
@@ -423,8 +428,14 @@ def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
             ax.set_title(darray._title_for_slice())
 
         if add_colorbar:
-            cbar = plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
-            if darray.name and add_labels:
+            cbar_kwargs = dict(cbar_kwargs)
+            cbar_kwargs.setdefault('extend', cmap_params['extend'])
+            if cbar_ax is None:
+                cbar_kwargs.setdefault('ax', ax)
+            else:
+                cbar_kwargs.setdefault('cax', cbar_ax)
+            cbar = plt.colorbar(primitive, **cbar_kwargs)
+            if darray.name and add_labels and 'label' not in cbar_kwargs:
```

Comment:
one corner case that may require checking is when `add_colorbar = False` and `cbar_ax` or `cbar_kwargs` is specified. It might be worth checking and just raising an error.  Unless you can think of a time when you want to be able to specify arguments that aren't used.